### DISCUSSION TITLE
Unable to install from PPA on Ubuntu 24.10 (Oracular)

### DISCUSSION THREAD

**Discussion by overbyte** [helix-editor/helix#11915] [General] [answered]
Attempted to use the PPA to install via apt on a brand new Ubuntu 24.10 install and received the following error after a `sudo apt update`

Apparently it's missing a release file?

```
➜  ~ sudo add-apt-repository ppa:maveonair/helix-editor
Repository: 'Types: deb
URIs: https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu/
Suites: oracular
Components: main
'
Description:
A post-modern text editor.

The source code used to build the package can be found here: https://github.com/maveonair/helix-ppa.
More info: https://launchpad.net/~maveonair/+archive/ubuntu/helix-editor
Adding repository.
Press [ENTER] to continue or Ctrl-c to cancel.

Hit:1 http://gb.archive.ubuntu.com/ubuntu oracular InRelease
Hit:2 http://gb.archive.ubuntu.com/ubuntu oracular-updates InRelease                                          
Hit:3 http://gb.archive.ubuntu.com/ubuntu oracular-backports InRelease                                        
Ign:4 https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular InRelease                       
Err:5 https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular Release
  404  Not Found [IP: 185.125.190.80 443]
Hit:6 http://security.ubuntu.com/ubuntu oracular-security InRelease
Reading package lists... Done
E: The repository 'https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

➜  ~ sudo apt update
Hit:1 http://gb.archive.ubuntu.com/ubuntu oracular InRelease
Hit:2 http://gb.archive.ubuntu.com/ubuntu oracular-updates InRelease                                          
Hit:3 http://gb.archive.ubuntu.com/ubuntu oracular-backports InRelease                                        
Ign:4 https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular InRelease                       
Err:5 https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular Release
  404  Not Found [IP: 185.125.190.80 443]
Hit:6 http://security.ubuntu.com/ubuntu oracular-security InRelease
Error: The repository 'https://ppa.launchpadcontent.net/maveonair/helix-editor/ubuntu oracular Release' does not have a Release file.
Notice: Updating from such a repository can't be done securely, and is therefore disabled by default.
Notice: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Comment by pascalkuthe** [comment, marked as answer]
The PPA is likely not updates yet for the new ubuntu release. We don't maintain the PPA (it's unofficial) so not actionable from our side

**Reply by overbyte** [reply]
ah okay - i've installed from snap for now - seems fine. Thanks for the response

**Reply by Youvegotmail101** [reply]
> ah okay - i've installed from snap for now - seems fine. Thanks for the response

What snap was it? 
(Sorry if this is a dumb question. This is my first attempt with Linux and I'm running into brick wall after brick wall. Is making me more determined to figure it out.. and also wanting to stop sucking at it) 

But I almost can't get anything to install. But my error message in terminal is verbatim the same as OP. 

ANY guidance is appreciated.

**Reply by David-Else** [reply]
@Youvegotmail101 Welcome to Linux, it is amazing!

In Linux you have official repositories (stores of software) for each Linux distribution. These are vetted by the people who make the distro to make sure they work and are free of anything dodgy. Depending on the distro they might be the latest versions, or maybe older stable versions that only get updated for security issues. On Ubuntu you get this software with a simple `sudo apt install <name of package>`.

PPA are user repositories, they are made by individual users and are not backed up by the people who make the distro or the people who make the software. They might be amazing or rubbish, you need to add them manually then use `apt install` and they will be picked up by the package manager, `apt`. Ideally the software is in the main distro repo so they are not needed.

In Ubuntu snaps are a controversial new type of packaging that only Ubuntu uses, Linux people don't like them much and prefer the traditional packaging system.

To get Helix working on the latest Ubuntu you can use the snap, the flatpak from flathub, write to the PPA owner and ask him to update it, compile from source, or download and run the Appimage supplied by the Helix crew https://github.com/helix-editor/helix/releases/download/24.07/helix-24.07-x86_64.AppImage. All these methods have small pros and cons. For example flatpak is better than Snap usually as it works on all Linux distros and is much more popular, but for a terminal app like Helix there are sandboxing complications for flatpak, so you would need to test it out to see if it works for you.

I use Debian 12 and compile Helix from source to get the latest features and help find any bugs. Debian 13 will have Helix included in the main repository.

I hope that gives you some insight into things.
